File: s3-inf-ppo/laboratoria/lab08/cs/lab08/Library.cs
using System;
using System.Collections.Generic;

namespace lab08
{
    class Library
    {
        public string Name { get; set; }
        public List<Book> Books { get; set; } = new List<Book>();

        public Library(string name)
        {
            Name = name;
        }

        public static Library operator +(Library library, Book book)
        {
            library.Books.Add(book);
            return library;
        }

        public void Print()
        {
            Console.WriteLine(Name);
            foreach (var book in Books)
                book.Print();
        }
    }
}

File: s3-inf-ppo/laboratoria/lab08/cpp/Library.cpp
#include "Library.h"

Library::Library(string name) {
	this->name = name;
}

Library& Library::operator +=(const Book &book) {
	this->books.push_back(book);
	return *this;
}

void Library::print() {
	cout << this->name << endl;
	for(int i = 0; i < this->books.size(); i++) {
		this->books.at(i).print();
	}
}
